core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -35,16 +36,32 @@ object Entity {
 
   // The type parameter aids type inference in Message constructors. This should be unnecessary when Messages are covariant.
   def empty[F[_]]: Entity[F] = Empty
+  def strict(chunk: Chunk[Byte]): Entity[Pure] = Strict(chunk)
 
   final case class Default[+F[_]](body: EntityBody[F], length: Option[Long]) extends Entity[F] {
     def ++[F1[x] >: F[x]](that: Entity[F1]): Entity[F1] = that match {
       case d: Default[F1] => Default(body ++ d.body, (length, d.length).mapN(_ + _))
+      case Strict(chunk) => Default(body ++ Stream.chunk(chunk), length.map(_ + chunk.size))
       case Empty => this
     }
 
     def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G] = Default(body.translate(fk), length)
   }
 
+  final case class Strict(chunk: Chunk[Byte]) extends Entity[Pure] {
+    val body: EntityBody[Pure] = Stream.chunk(chunk)
```

Comment:
`body` then could be `stream`, maybe.

Reply:
These names are currently inherited from `Entity` itself. Do you think we should rename them for everything?
```scala
sealed trait Entity[+F[_]] {
  def body: EntityBody[F]
  def length: Option[Long]
  def ++[F1[x] >: F[x]](that: Entity[F1]): Entity[F1]
  def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G]
}
```

Reply:
We can encapsulate the type of `body` to do it, like
```scala
abstract sealed trait Entity[+F[_]] {
  type EntityBodyType
  def body: EntityBodyType
  def length: Option[Long]
  def ++[F1[x] >: F[x]](that: Entity[F1]): Entity[F1]
  def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G]
}
```

Reply:
Huh, that's an interesting idea. How does somebody given an `Entity` use the body if they don't know the type?

Reply:
If we'll have `def stream: EntityBody[F]` I see no problem with that :)

Reply:
So, you can only use `body` if you know the subclass of `Entity`, otherwise you use `stream` 👍 I'm a little confused then, why do we need to define `body` as part of the shared `Entity` interface then? Since it's very difficult to use unless you know the specific subclass of `Entity`. Instead, we could just use the `body` name on each subclass. WDYT?

Reply:
I agree with renaming `body` to `stream` (or maybe `byteStream`?) - not sure about renaming `chunk` to `body`, since the latter is less informative, but I don't have a very strong opinion on this. Another idea would be to call it `toChunk`, which doesn't carry any implication about the internal representation.

Reply:
I agree with @armanbilge that there doesn't seem much benefit adding a `BodyType` type member, but I'm also not keen on having a bunch of different `body` members with different types. It seems confusing.

Reply:
I like `byteStream` and `toChunk` 👍

Reply:
Well, I'm not sure actually. Does it make sense to `toChunk` an _entity_? An entity body for sure can be represented as a `Chunk`, but I'm not sure about `Entity` itself 🤔

Reply:
> what about just bytes?

But the chunk is *also* the bytes.

So maybe just `stream` and `chunk`? Or `byteStream` and `byteChunk`?

Reply:
😆 

I kind of want to leave it the way it is. Isn't `entity-body` the "official" name for this anyway?
```
entity-body    = *OCTET
```
https://www.w3.org/Protocols/rfc2616/rfc2616-sec7.html

Reply:
Fine by me 😆